test/box-luatest/gh_7897_sending_arbitrary_iproto_packets_test.lua

```diff
@@ -0,0 +1,224 @@
+                   if mp_body ~= nil then
+                       packet_body, next = msgpack.decode(mp, next)
+                   end
+                   t.assert_equals(next, mp_len + 1)
+                   t.assert_equals(packet_header, lua_header)
+                   t.assert_equals(packet_body, lua_body)
+               end)
+    end)
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.before_each(function(cg)
+    cg.server:exec(function(net_box_uri)
+        local uri = require('uri')
+        local socket = require('socket')
+        local t = require('luatest')
+
+        -- Connect to the server.
+        local u = uri.parse(net_box_uri)
+        local s = socket.tcp_connect(u.host, u.service)
+        t.assert_is_not(s, nil)
+        -- Skip the greeting.
+        t.assert_equals(#s:read(128), 128)
+        rawset(_G, 's', s)
+    end, {cg.server.net_box_uri})
+end)
+
+-- Checks that `box.iproto.send` errors are handled correctly.
+g.test_box_iproto_send_errors = function(cg)
+    cg.server:exec(function()
+        local fiber = require('fiber')
+        local t = require('luatest')
+
+        -- Checks that invalid argument counts are handled correctly.
+        local usage_err_msg = "Usage: box.iproto.send(sid, header[, body])"
+        t.assert_error_msg_content_equals(usage_err_msg,
+                                          function() box.iproto.send() end)
+        t.assert_error_msg_content_equals(usage_err_msg,
+                                          function() box.iproto.send(0) end)
+        -- Checks that invalid session identifiers are handled correctly.
+        t.assert_error_msg_content_equals("expected uint64_t as 1 argument",
+                                          function()
+                                              box.iproto.send('str', {})
+                                          end)
+        t.assert_error_msg_content_equals("Session 777 does not exist",
+                                          function()
+                                              box.iproto.send(777, {})
+                                          end)
+        -- Checks that invalid session type is handled correctly.
+        local bg
+        fiber.create(function() bg = box.session.id(); fiber.yield() end)
+        t.assert_error_msg_content_equals(
+                "Session 'background' is not supported",
+                function() box.iproto.send(bg, {}) end)
+        -- Checks that invalid 2 argument type is handled correctly.
+        t.assert_error_msg_content_equals(
+                "expected table or string as 2 argument",
+                function() box.iproto.send(0, 0) end)
+        -- Checks that invalid 3 argument type is handled correctly.
+        t.assert_error_msg_content_equals(
+                "expected table or string as 3 argument",
+                function() box.iproto.send(0, {}, 0) end)
+
+        local sid = rawget(_G, 'sid')
+        local s = rawget(_G, 's')
```

Comment:
Nit (up to you): all those `rawget(_G, 'key')` could be replaced with `_G.key`. It won't throw an error if the key is set.

Reply:
I didn't think about it, thanks for pointing this out, fixed!